openedx/features/discounts/applicability.py

```diff
@@ -73,8 +73,16 @@ def get_discount_expiration_date(user, course):
                 content_availability_date = enrollment.created
     except CourseEnrollment.schedule.RelatedObjectDoesNotExist:
         content_availability_date = max(enrollment.created, course.start)
+    discount_expiration_date = content_availability_date + timedelta(weeks=1)
 
-    return content_availability_date + timedelta(weeks=1)
+    verified_mode = CourseMode.verified_mode_for_course(course=course, include_expired=True)
```

Comment:
I know why we are doing this, but I am not sure I would know as a casual reader of the code. Do you mind commenting here what we are doing and why?